torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -373,7 +371,7 @@ bool FunctionSignature::parse(PyObject* args, PyObject* kwargs, PyObject* dst[],
 
     if (!is_kwd) {
       arg_pos++;
-    } else {
+    } else if (obj) {
```

Comment:
which case does this fix?

Reply:
We we're incorrectly matching some signatures because we decremented `remaining_kwargs` even when there wasn't a kwarg. For example, from the tests:

```
x.new(sizes=(1, 2, 3))
```

Was matching:

```
Tensor(*, int64_t device=-1)
```

because device is optional and we incorrectly decremented remaining_kwargs to 0 after we set `device` to it's default value (-1).